common/src/main/java/com/microsoft/identity/common/components/AndroidStorageLoader.java

```diff
@@ -0,0 +1,142 @@
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.components;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import androidx.annotation.Nullable;
+
+import com.microsoft.identity.common.internal.cache.SharedPreferencesFileManager;
+import com.microsoft.identity.common.internal.util.SharedPrefStringNameValueStorage;
+import com.microsoft.identity.common.internal.util.SharedPreferenceLongStorage;
+import com.microsoft.identity.common.java.cache.IMultiTypeNameValueStorage;
+import com.microsoft.identity.common.java.crypto.IKeyAccessor;
+import com.microsoft.identity.common.java.interfaces.INameValueStorage;
+import com.microsoft.identity.common.java.interfaces.IStorageLoader;
+import com.microsoft.identity.common.java.util.ported.Predicate;
+
+import java.util.Iterator;
+import java.util.Map;
+
+import lombok.AllArgsConstructor;
+import lombok.NonNull;
+
+@AllArgsConstructor
+public class AndroidStorageLoader implements IStorageLoader {
+    private final Context mContext;
+
+    @Override
+    @NonNull
+    public <T> INameValueStorage<T> getNameValueStore(@NonNull final String storeName,
+                                                      @NonNull final Class<T> clazz) {
+        return getEncryptedNameValueStore(storeName, null, clazz);
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    @NonNull
+    public <T> INameValueStorage<T> getEncryptedNameValueStore(@NonNull final String storeName,
+                                                               @Nullable final IKeyAccessor helper,
+                                                               @NonNull final Class<T> clazz) {
+        final IMultiTypeNameValueStorage mgr = SharedPreferencesFileManager.getSharedPreferences(mContext, storeName, helper);
+        if (Long.class.isAssignableFrom(clazz)) {
+            return (INameValueStorage<T>) new SharedPreferenceLongStorage(mgr);
+        } else if (String.class.isAssignableFrom(clazz)) {
+            return (INameValueStorage<T>) new SharedPrefStringNameValueStorage(mgr);
+        }
+        throw new UnsupportedOperationException("Only Long and String are natively supported as types");
+    }
+
+    @Override
+    @NonNull
+    public IMultiTypeNameValueStorage getEncryptedFileStore(@NonNull final String storeName,
+                                                            @NonNull final IKeyAccessor helper) {
+        return SharedPreferencesFileManager.getSharedPreferences(mContext, storeName, helper);
+    }
```

Comment:
This can move to `IStorageLoader` as `default` implementation - that way this can be leveraged in Linux as well.